src/gui/qgsmapcanvas.h

```diff
@@ -236,6 +237,11 @@ class GUI_EXPORT QgsMapCanvas : public QGraphicsView
       @param layer optionally specify different than current layer */
     void zoomToSelected( QgsVectorLayer* layer = nullptr );
 
+    /** Set canvas extent to the bounding box of a feature
+        @param layer the vector layer
+        @param id the feature id*/
+    void zoomToFeatureId( QgsVectorLayer* layer, QgsFeatureId id );
```

Comment:
There is still the possibility to zoom to a given extent for plugins. Anyway, I'm merging the PR as is now. Further enhancements can still be added in the future.